src/main/java/io/supertokens/webserver/api/oauth/OAuthClientsAPI.java

```diff
@@ -0,0 +1,168 @@
+        if(!containsAllRequired || containsMoreThanAllowed){
+            throw new ServletException(new WebserverAPI.BadRequestException("Invalid Json Input"));
+        }
+
+        try {
+            AppIdentifier appIdentifier = getAppIdentifier(req);
+            Storage storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+
+            JsonObject response = OAuth.registerOAuthClient(super.main, appIdentifier, storage, input);
+            sendJsonResponse(200, response, resp);
+
+        } catch (OAuthClientRegisterException registerException) {
+
+            JsonObject errorResponse = createJsonFromException(registerException);
+            sendJsonResponse(400, errorResponse, resp);
+
+        } catch (TenantOrAppNotFoundException | InvalidConfigException | BadPermissionException
+                 | NoSuchAlgorithmException | StorageQueryException e) {
+            throw new ServletException(e);
+        }
+    }
+
+    @Override
+    protected void doGet(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        String clientId = InputParser.getQueryParamOrThrowError(req, "clientId", false);
+
+        try {
+            AppIdentifier appIdentifier = getAppIdentifier(req);
+            Storage storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+
+            JsonObject response = OAuth.loadOAuthClient(main, appIdentifier, storage, clientId);
+            sendJsonResponse(200, response, resp);
+
+        }  catch (OAuthClientException e) {
+            JsonObject errorResponse = createJsonFromException(e);
+            sendJsonResponse(400, errorResponse, resp);
+
+        } catch (TenantOrAppNotFoundException | InvalidConfigException | BadPermissionException
+                 | StorageQueryException e){
+            throw new ServletException(e);
+        }
+    }
+
+    @Override
+    protected void doDelete(HttpServletRequest req, HttpServletResponse resp) throws IOException, ServletException {
+        JsonObject requestBody = InputParser.parseJsonObjectOrThrowError(req);
+
+        String clientId = InputParser.parseStringOrThrowError(requestBody, "clientId", false);
+
+        try {
+            AppIdentifier appIdentifier = getAppIdentifier(req);
+            Storage storage = enforcePublicTenantAndGetPublicTenantStorage(req);
+
+            OAuth.deleteOAuthClient(main, appIdentifier, storage, clientId);
+            JsonObject responseBody = new JsonObject();
+            responseBody.addProperty("status", "OK");
+            sendJsonResponse(200, responseBody, resp);
+
+        }  catch (OAuthClientException e) {
+            JsonObject errorResponse = createJsonFromException(e);
+            sendJsonResponse(400, errorResponse, resp);
+
+        } catch (TenantOrAppNotFoundException | InvalidConfigException | BadPermissionException
+                 | StorageQueryException e){
+            throw new ServletException(e);
+        }
+    }
+
```

Comment:
API to modify an existing client id?

Reply:
I didn't get to it yesterday, working on it today :)

Reply:
I dont see this in the PR either. Maybe you forgot to push?

Reply:
No, I didn't got to it. That will be in a different PR